File: src/main/java/dev/alexengrig/leetcode/_141_linked_list_cycle/Task.java
package dev.alexengrig.leetcode._141_linked_list_cycle;

interface Task {
    boolean hasCycle(ListNode head);

    class ListNode {
        int val;
        ListNode next;

        ListNode(int x) {
            val = x;
            next = null;
        }

        ListNode(int x, ListNode node) {
            val = x;
            next = node;
        }
    }
}

File: src/main/python/138_copy_list_with_random_pointer/task.py
from typing import Optional


class Node:
    def __init__(self, x: int, next: Optional["Node"] = None, random: Optional["Node"] = None):
        self.val = int(x)
        self.next = next
        self.random = random


class Task:
    def copyRandomList(self, head: 'Optional[Node]') -> 'Optional[Node]':
        pass
